Add handling for empty error object in deploy.ts

Gabulhas opened the pull request:
Super simple, but was facing some problems here for example

https://ao_marton.g8way.io/#/message/K5bEfUQh6Cc_bnyf41uJi8eWGfAJ6G5w2-LW-oij7Rk


where the deployment was getting an empty "Error" attribute, like the following:
```
{
  "Messages": [],
  "Assignments": [],
  "Spawns": [],
  "Output": "",
  "Error": {}      <-- It was breaking because of this
} 
```

Gabulhas committed:
Add handling for empty error object in deploy.ts

pawanpaudel93 commented:
@Gabulhas Thanks for the PR. Can you please run `pnpm lint --fix` and push the changes ?

Gabulhas committed:
linting

pawanpaudel93 commented on src/lib/deploy.ts:
```diff
@@ -146,7 +146,14 @@ export class DeploymentsManager {
+    if (Output && Output.data && Output.data.output) {
+      errorMessage = Output.data.output
+    }
+    else if (error && Object.keys(error).length > 0) {
```
I have seen `Error` to be string most of the times if present and it is absent while no error exists. But now we know it can be `Object` too. The code works but lets convert error of type `Object` to string while assigning it to `errorMessage`.

`error && Object.keys(error).length > 0` Also can this check be modified to support other data types too ?

Gabulhas replied:
sorry for the long reply/fix
let me know if this works

pawanpaudel93 replied:
I think it should work. Seems you deleted `pnpm-lock.yaml`

Gabulhas replied:
fixed, can't remember why I deleted, probably was a misclick on my editor

Gabulhas committed:
fix: handle error as string or object in errorMessage assignment

- Convert error to string if it's an object using JSON.stringify()
- Use String() for other data types
- Simplify error check to handle various data types

Gabulhas committed:
Re-added pnpm-lock.yaml

pawanpaudel93 commented:
@Gabulhas Thanks for the PR. Its merged and new version is published.
